jerry-core/ecma/operations/ecma-regexp-object.cpp

```diff
@@ -257,13 +257,6 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
 
   while ((op = re_get_opcode (&bc_p)))
   {
-    if (re_ctx_p->match_limit >= RE_EXECUTE_MATCH_LIMIT)
-    {
-      ret_value = ecma_raise_range_error ("RegExp executor steps limit is exceeded.");
-      return ret_value;
-    }
-    re_ctx_p->match_limit++;
```

Comment:
I think the match limit counter should be removed from the executor context as well, since we don't use it anymore.